src/custom/api/gnosisProtocol/api.ts

```diff
@@ -276,7 +281,45 @@ async function _handleQuoteResponse(response: Response, params?: FeeQuoteParams)
   }
 }
 
-export async function getPriceQuote(params: PriceQuoteParams): Promise<PriceInformation | null> {
+function _mapNewToLegacyParams(params: FeeQuoteParams): QuoteQuery {
+  const { amount, kind, userAddress, receiver, validTo, sellToken, buyToken } = params
+  const fallbackAddress = userAddress || ZERO_ADDRESS
+
+  const baseParams = {
+    sellToken,
+    buyToken,
+    from: fallbackAddress,
+    receiver: receiver || fallbackAddress,
+    appData: getAppDataHash(),
+    validTo,
+    partiallyFillable: false,
+  }
+
+  const finalParams: QuoteQuery =
+    kind === OrderKind.SELL
+      ? {
+          kind: OrderKind.SELL,
+          sellAmountBeforeFee: amount,
+          ...baseParams,
+        }
+      : {
+          kind: OrderKind.BUY,
+          buyAmountAfterFee: amount,
+          ...baseParams,
+        }
```

Comment:
u can simplify and refactor above: kind is already a var with the right value, ...baseParams expansion are done in both cases, so the only difference is the name of the prop where you place the amount.

so make that the only difference
const amountFiled = kind === OrderKind.SELL ? 'sellAmountBeforeFee' : 'buyAmountAfterFee'

Reply:
TS will complain

Reply:
It didn't to me https://github.com/gnosis/cowswap/pull/1847

Reply:
it did

Reply:
Damm typescript!